### PR TITLE
Add DefaultShardManagerBuilder(String) constructor

### PR DESCRIPTION
**Pull request by BenjaminUrquhart**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [X] Library interface (affecting end-user code) 
- [ ] Other: \_____ 
- [ ] Documentation

Closes Issue: NaN

## Description

Added a constructor to `DefaultShardManager` to mirror `JDABuilder(String)`.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java [hunk 1]
```diff
@@ -82,6 +82,18 @@
      * before calling {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#build() build()}.
      */
     public DefaultShardManagerBuilder() {}
+    
+    /**
+     * Creates a nearly empty DefaultShardManagerBuilder with the given token.
+     * <br>This is equivelent to using the constuctor
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: equivalent*

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java [hunk 2]
```diff
@@ -82,6 +82,18 @@
...
+     * {@link net.dv8tion.jda.bot.sharding#DefaultShardManagerBuilder() DefaultShardManager()}
+     * and calling {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#setToken(String) setToken(String)}
+     * directly afterward.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Mention that you can change the token afterwards by using `setToken` again

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java [hunk 3]
```diff
@@ -82,6 +82,18 @@
      * before calling {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#build() build()}.
      */
     public DefaultShardManagerBuilder() {}
+    
+    /**
+     * Creates a nearly empty DefaultShardManagerBuilder with the given token.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Why the "nearly empty"? Seems redundant to me

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java [hunk 4]
```diff
@@ -82,6 +82,19 @@
      * before calling {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#build() build()}.
      */
     public DefaultShardManagerBuilder() {}
+    
+    /**
+     * Creates a DefaultShardManagerBuilder with the given token.
+     * <br>This is equivalent to using the constuctor
+     * {@link net.dv8tion.jda.bot.sharding#DefaultShardManagerBuilder() DefaultShardManager()}
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Is this link actually working? Shouldn't this be `...sharding.DefaultShardManagerBuilder#DefaultShar...()`?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Also the alias is wrong

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java [hunk 5]
```diff
@@ -82,6 +82,19 @@
...
+     * and calling {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#setToken(String) setToken(String)}
+     * directly afterward. You can always change the token later with 
+     * {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#setToken(String) setToken(String)}.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: just do `{@link #setToken(String) ...}` on both of these

Comment (BenjaminUrquhart) [reply src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Done

File: src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java (hunk 4 above)
Comment (BenjaminUrquhart) [reply src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java]: Fixed the alias. In theory, the constructor link works, haven't had a chance to try it out (I've been working from the web editor)

**Pull request merged**